cms/static/js/views/license-selector.js

```diff
@@ -0,0 +1,107 @@
+/**
+ * Provide license selector view for license models.
+ * These license selectors are used to set the license for
+ * new or existing courses and videos.
+ */
+define(["js/views/baseview", "underscore", "gettext", "js/models/license", "js/views/license-view"],
+    function(BaseView, _, gettext, LicenseModel, LicenseView) {
+
+        var LicenseSelector = BaseView.extend({
+            events: {
+                "click .license-button" : "onLicenseButtonClick",
+            },
+
+            initialize: function(options) {
+                this.template = this.loadTemplate("license-selector");
+                if (!this.model) {
+                    this.model = new LicenseModel();
+                }
+                else if (!(this.model instanceof LicenseModel) && this.model.kind) {
+                    this.model = new LicenseModel(this.model);
+                }
```

Comment:
Whoa whoa whoa. This is going to be *incredibly* confusing for anyone trying to debug this code. You pass in a model, and the view uses a different model than the one you passed in? That's going to cause all sorts of unexpected situations and behaviors.

Instead, I recommend one of two approaches. I think the best option is to just ignore this type check, and assume that the model has the attributes that this view is expecting. This is called [duck typing](https://en.wikipedia.org/wiki/Duck_typing), and it works surprisingly well. It also means that people can write new models that match the `LicenseModel` API, plug them into this view, and they will Just Work. If you don't want to use duck typing, then I'd suggest that you simply throw an error if the model that is passed in is not the model that you're expecting. Let the developer deal with that by wrapping the model herself before passing it to the view, so that she knows that the wrapping is happening.